### PR TITLE
Website 0.10

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

File: website/versioned_docs/version-0.10/clientsidequeries.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+id: version-0.10-clientqueries
+title: Client-side Queries
+original_id: clientqueries
+---
+
+Graphback allows users to generate client-side queries based on the data model. All the queries/mutations/subscriptions
+which are generated through `graphback generate` for the user can also be generated as client-side queries. Users may follow up with using `create-react-app` or `apollo-client` to create client.
+
+## How it works
+Graphback asks whether you want to generate client-side queries in the `graphback init` command. Answering it yes,
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/clientsidequeries.md]:
```suggestion
Graphback asks whether you want to generate client-side queries in the `graphback init` command. Answering it yes will set `client` in `config.json` to `true`.
```

Comment (craicoverflow) [inline website/versioned_docs/version-0.10/clientsidequeries.md]: `graphback init` command is gone now.

Comment (craicoverflow) [reply website/versioned_docs/version-0.10/clientsidequeries.md]: Should `config.json` be `graphback.json`?

File: website/versioned_docs/version-0.10/clientsidequeries.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+For a data model having
+```
+type Note {
+  id: ID!
+  title: String!
+  description: String!
+}
+```
+running `graphback generate` will generate all operations which are enabled in `config.json` or using `annotations`.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/clientsidequeries.md]:
```suggestion
running `graphback generate` will generate all operations which are enabled in `graphback.json` or using `annotations`.
```

Should `config.json` be `graphback.json`?

File: website/versioned_docs/version-0.10/clientsidequeries.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  ${CommentFragment}
+`
+
+```
+
+## Autogenerating forms based on server-side schema
+
+For react based applications developers can use [uniform.tools](https://uniforms.tools).
+Uniforms can be used along with generated client queries to automatically build dynamic forms with labales and validation.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/clientsidequeries.md]:
```suggestion
Uniforms can be used along with generated client queries to automatically build dynamic forms with labels and validation.
```

File: website/versioned_docs/version-0.10/commands.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+### Config 
+
+```bash
+graphback config
+```
+
+Create configuration in existing project to support Graphback resolver and schema generation.
+
+> **Note** When creating configuration please review if locations created for `files` integration match your project folder structure
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/commands.md]:
```suggestion
> **Note** When creating configuration please review if locations created for `folders` integration match your project folder structure
```

Did you mean `folders` here?

File: website/versioned_docs/version-0.10/commands.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  graphback generate
+  ```
+  Generate command will generate schema and resolvers for your GraphQL server based on your data model.
+
+### DB
+  ```bash
+  graphback db
+  ```
+  Create databases resources based on your data model. Currently supports PostgreSQL and sqlite3.
+> **Note** - *Running `graphback db` drops your database and recreates it with tables and relationships.*
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/commands.md]: This is no longer the case.

File: website/versioned_docs/version-0.10/commands.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+## Update Database
+
+```sh
+graphback update-db
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/commands.md]: This command is gone now. Everything is done in `graphback db`

File: website/versioned_docs/version-0.10/commands.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+Updates your database schema based on your data model. Currently supported change types are:
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/commands.md]: This limitation is based on `migrateDBUsingSchema` which is not used by the CLI.

File: website/versioned_docs/version-0.10/concepts.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+- `./model` folder will contain input schema
+- `./resolvers` folder will contain generated and custom GraphQL resolvers for the project.
+- `./client` folder will contain client side queries
+
+All locations can be changed in `graphback.json` configuration file in your project root
+
+## Runtime CRUD layer
+
+Code generators produce a large amount of the code that needs to be maintained later.
+Having generator code diverging from original form will prevent developers to utilize generators at later stage.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/concepts.md]:
```suggestion
Having generator code diverging from original form will prevent developers utilizing generators at a later stage. 
```

File: website/versioned_docs/version-0.10/concepts.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+- `./client` folder will contain client side queries
+
+All locations can be changed in `graphback.json` configuration file in your project root
+
+## Runtime CRUD layer
+
+Code generators produce a large amount of the code that needs to be maintained later.
+Having generator code diverging from original form will prevent developers to utilize generators at later stage. 
+Additionally writing your custom handlers is hard as you will always need to start from scratch and benefit from the existing layers that provide support for subscriptions and data access. 
+Graphback utilizes runtime layer that provide data access methods with support for extensibility.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/concepts.md]:
```suggestion
Graphback utilizes runtime layer that provides data access methods with support for extensibility.
```

File: website/versioned_docs/version-0.10/concepts.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
+---
+id: version-0.10-concepts
+title: Introduction
+original_id: concepts
+---
+
+## Introduction
+
+The target of Graphback is to provide tooling for generating fully functional backend and client side code on top of your existing application and widely available starter applications.
+Graphback making prototyping applications very easy by generating source code and reducing number of boilerplate components. With Graphback developers can focus more on delivering business value and their business model that will be represented as GraphQL schema. Developers can choose different starting stack database and extensions to build their production ready backend in a very short amount of time.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/concepts.md]:
```suggestion
Graphback makes prototyping applications very easy by generating source code and reducing number of boilerplate components. With Graphback developers can focus more on delivering business value and their business model that will be represented as GraphQL schema. Developers can choose different starting stack database and extensions to build their production ready backend in a very short amount of time.
```

File: website/versioned_docs/version-0.10/concepts.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+Graphback generates codebase that can be edited and customized.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/concepts.md]:
```suggestion
Graphback generates a codebase that can be edited and customized.
```

File: website/versioned_docs/version-0.10/concepts.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+Code generators produce a large amount of the code that needs to be maintained later.
+Having generator code diverging from original form will prevent developers to utilize generators at later stage. 
+Additionally writing your custom handlers is hard as you will always need to start from scratch and benefit from the existing layers that provide support for subscriptions and data access. 
+Graphback utilizes runtime layer that provide data access methods with support for extensibility.
+
+For more information please refer to [`runtime`](/docs/runtime) documentation
+
+## Database migrations
+
+To offer fully comprehensive getting started experience Graphback provides database migration capabilities.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/concepts.md]:
```suggestion
To offer a fully comprehensive getting started experience Graphback provides database migration capabilities. 
```

File: website/versioned_docs/version-0.10/config.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+We can use them in clients to receive live updates for every change that is happening on the server.
+
+ Graphback allows the user to customize the generation process by using `configuration`.
+
+## Configuration
+Graphback uses flags to allow user to choose between the CRUD operations and subscriptions. These are present in `graphback.json`, created by `config` command or by graphql-cli,
+in the root of your project folder under the `generation` key. The default config is
+```json
+{
+  ...,
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/config.md]:
```suggestion
  ...
```

File: website/versioned_docs/version-0.10/config.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    "subDelete": false,
+    "disableGen": false,
+  }
+}
+```
+Changing these flags and performing `generate`, regenerates your `schema` and `resolvers` with provided config.
+> **Note**: For subscriptions, the user needs to change the value of the respective operations to `true`. For example, changing
+`subDelete` to `true` won't work unless, `delete` is `true`.
+
+## Changing configuration for the type
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/config.md]:
```suggestion
## Changing configuration per type
```

File: website/versioned_docs/version-0.10/config.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+  ...
+}
+```
+will create the `delete` mutation for `Note` type.
+
+> **Note**: Annotations override the configuration flags to `true`
+
+#### `@crud.disableGen`
+
+User can use this directive to disable CRUD operation generation for that type. Applying this directive will not create any
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/config.md]:
```suggestion
User can use this annotation to disable CRUD operation generation for that type. Applying this annotation will not create any 
```

File: website/versioned_docs/version-0.10/config.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+}
+```
+will create the `delete` mutation for `Note` type.
+
+> **Note**: Annotations override the configuration flags to `true`
+
+#### `@crud.disableGen`
+
+User can use this directive to disable CRUD operation generation for that type. Applying this directive will not create any 
+`Query`/`Mutation`/`Subscription` for that type.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/config.md]: I think this should be kept on the same line as the rest of the sentence.

File: website/versioned_docs/version-0.10/database-schema-migrations.md [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+  note: Note! # this creates a `noteId` column in the `comment` table.
+}
+
+type Note {
+  id: ID!
+  title: String!
+}
+```
+
+> NOTE: The field name must match that of the related type. Support for custom field names is coming soon.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/database-schema-migrations.md]: This is out of date. This one is on me.

I've provided a PR o update this here https://github.com/aerogear/graphback/pull/609

File: website/versioned_docs/version-0.10/database-schema-migrations.md [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+  id: ID!
+  """
+  @db.type: 'string'
+  @db.length: 100
+  """
+  title: String!
+}
+```
+
+#### Changing column names in graphback
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/database-schema-migrations.md]:
```suggestion
#### Changing column names in Graphback
```

File: website/versioned_docs/version-0.10/database-schema-migrations.md [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+type Note {
+  id: ID!
+  """
+  @db.name: 'note_title'
+  """
+  title: String!
+}
+```
+
+When using custom name in database we need to map it directly inside resolver or db layer.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/database-schema-migrations.md]:
```suggestion
When using a custom name in the database we need to map it directly inside resolver or db layer.
```

File: website/versioned_docs/version-0.10/database-schema-migrations.md [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+
+> NOTE: database migration logic require objects to have `id: ID!` field defined. 
+Renaming `id` field to anything else will break Graphback data resolution logic
+
+### Compatibility
+
+The following database providers support full database schema migrations.
+
+- PostgreSQL
+- SQLLite
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/database-schema-migrations.md]:
```suggestion
- SQLite
```

File: website/versioned_docs/version-0.10/gettingstarted.md [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+    - GQL (for Schema)
+
+## Multiple ways of getting started
+
+You can start with Graphback by
+
+- Using Graphback with most popular GraphQL boilerplates with GraphQL CLI
+- Adding Graphback to your existing project
+- Initializing Graphback in your code without code generation
+- Migrating your existing RESTfull api to GraphQL
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/gettingstarted.md]:
```suggestion
- Migrating your existing REST API to GraphQL 
```

File: website/versioned_docs/version-0.10/gettingstarted.md [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+You can start with Graphback by
+
+- Using Graphback with most popular GraphQL boilerplates with GraphQL CLI
+- Adding Graphback to your existing project
+- Initializing Graphback in your code without code generation
+- Migrating your existing RESTfull api to GraphQL 
+
+### Start new project using Graphback and GraphQL CLI
+
+Best way to start with Graphback is to choose one of the available project boilerplates
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/gettingstarted.md]:
```suggestion
The best way to start with Graphback is to choose one of the available project boilerplates 
```

File: website/versioned_docs/version-0.10/gettingstarted.md [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+- Using Graphback with most popular GraphQL boilerplates with GraphQL CLI
+- Adding Graphback to your existing project
+- Initializing Graphback in your code without code generation
+- Migrating your existing RESTfull api to GraphQL 
+
+### Start new project using Graphback and GraphQL CLI
+
+Best way to start with Graphback is to choose one of the available project boilerplates 
+available as part of official GraphQL CLI. 
+GraphQL CLI integrates Graphback with other community libraries offering compehensive
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/gettingstarted.md]:
```suggestion
GraphQL CLI integrates Graphback with other community libraries offering comprehensive 
```

typo

File: website/versioned_docs/version-0.10/gettingstarted.md [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+- Adding Graphback to your existing project
+- Initializing Graphback in your code without code generation
+- Migrating your existing RESTfull api to GraphQL 
+
+### Start new project using Graphback and GraphQL CLI
+
+Best way to start with Graphback is to choose one of the available project boilerplates 
+available as part of official GraphQL CLI. 
+GraphQL CLI integrates Graphback with other community libraries offering compehensive 
+way to work with GraphQL API's.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/gettingstarted.md]:
```suggestion
way to work with GraphQL APIs.
```

File: website/versioned_docs/version-0.10/gettingstarted.md [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+```
+Follow the onscreen instructions.
+
+You can use Graphback with either a SQLite (for prototyping) or PostgreSQL (for production) databases.
+
+If you use SQLite, you don't need to use Docker.
+
+## Running Graphback in existing application without code generation
+
+Graphback offers runtime layer that will create in memory implementation of the graphql services without code generation.
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/gettingstarted.md]:
```suggestion
Graphback offers runtime layer that will create an in-memory implementation of the GraphQL services without code generation.
```

File: website/versioned_docs/version-0.10/releases.md [hunk 26]
```diff
@@ -0,0 +1,269 @@
...
+Graphback no longer uses GraphQL directives for controling generation of the CRUD operations.
+Users should use annotations
+
+```
+type Note @delete {
+  ...
+}
+```
+
+becomes now:
```
Comment (craicoverflow) [inline website/versioned_docs/version-0.10/releases.md]:
```suggestion
now becomes:
```

**Comment by wtrocki** [comment]
Hi

This change is just generating new version based on master. 
We should apply this change first on the master and then regenerate new version. 
I will take look on this in free time

**Comment by wtrocki** [comment]
Note that changes were already published

**Pull request closed without merging**